File: 2201-count-artifacts-that-can-be-extracted/2201-count-artifacts-that-can-be-extracted.cs
public class Solution {
    public int DigArtifacts(int n, int[][] artifacts, int[][] dig) {
        var di = new bool[n,n];
        var extract = 0;
        
        // in nxn matrix, mark all positions which are digged
        foreach(var d in dig)
            di[d[0], d[1]] = true;
        
        // for each artifact, check if required positions are digged or not
        foreach(var a in artifacts){
            var found = true;
            for(int i=a[0]; i<=a[2]; i++){
                for(int j=a[1]; j<=a[3]; j++){
                    if(!di[i,j]){
                        found = false;
                        break;
                    }
                }
                if(!found) break;
            }
            if(found) extract++;
        }
        
        return extract;
    }
}

File: 2201-count-artifacts-that-can-be-extracted/2201-count-artifacts-that-can-be-extracted.py
class Solution:
    def digArtifacts(self, n: int, artifacts: List[List[int]], dig: List[List[int]]) -> int:
        digged = [[False]*n for _ in range(n)]
        for dg in dig:
            digged[dg[0]][dg[1]] = True
        
        extract = 0
        for artifact in artifacts:
            x1,y1,x2,y2 = artifact[0], artifact[1], artifact[2], artifact[3]
            found = True
            for i in range(x1, x2+1):
                for j in range(y1, y2+1):
                    if not digged[i][j]:
                        found = False
                        break
                if not found: break
            
            if found: extract += 1
        
        return extract
                    